pkg/kubelet/apis/config/types.go

```diff
@@ -229,6 +229,10 @@ type KubeletConfiguration struct {
 	CgroupsPerQOS bool
 	// driver that the kubelet uses to manipulate cgroups on the host (cgroupfs or systemd)
 	CgroupDriver string
+	// SingleProcessOOMKill, if true, will prevent the `memory.oom.group` flag from being set for container
+	// cgroups in cgroups v2. This causes processes in the container to be OOM killed individually instead of as
+	// a group. It means that if true, the behavior aligns with the behavior of cgroups v1.
+	SingleProcessOOMKill bool
```

Comment:
Could you update the internal type to use `*bool` so that we can properly distinguish `false` and `nil`, and validate based on that? See [this discussion thread](https://github.com/kubernetes/kubernetes/pull/126096#discussion_r1746055794) for more details.